core/vm/evm.go

```diff
@@ -602,6 +604,86 @@ func (evm *EVM) CallFromSystem(contractAddress common.Address, abi abipkg.ABI, f
 	return evm.handleABICall(abi, funcName, args, returnObj, call)
 }
 
+type cacheKey struct {
```

Comment:
We need to include `gas` in this key as it can effect the execution. It shouldn't reduce our hit rate because we use the same gas value every time.